pkg/kinflate/resource/resource.go

```diff
@@ -16,10 +16,41 @@ limitations under the License.
 
 package resource
 
-import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+import (
+	"encoding/json"
 
-// Examples: deployment, service, configmap, secret, etc.
-type Resource interface {
-	Name() string
-	Data() unstructured.Unstructured
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/kinflate/types"
+)
+
+// Resource represents a Kubernetes Resource Object for ex. Deployment, Server
+// ConfigMap etc.
+type Resource struct {
+	Data *unstructured.Unstructured
```

Comment:
Do we need a pointer?

Reply:
I believe we do. Unstructured is a struct instead of an interface.

Reply:
That doesn't necessarily means you should have a pointer though, does it? :-)
I'm curious if there's a rule for that ...

Reply:
One thing to note here is having it as struct doesn't help much because it has map[string]interface{} as a field inside where all the data resides and that is NOT copied when you pass the struct around. So technically you are NOT working on a copy.